src/main/scala/cromwell/binding/command/ParameterCommandPart.scala

```diff
@@ -3,17 +3,19 @@ package cromwell.binding.command
 import cromwell.binding.WdlValue
 import cromwell.binding.types.WdlType
 
+import scala.util.Try
+
 case class ParameterCommandPart(wdlType: WdlType, name: String) extends CommandPart {
   override def toString: String = "${" + s"$wdlType $name" + "}"
 
-  def instantiate(parameters: Map[String, WdlValue]): String = {
-    val paramValue = parameters.getOrElse(name, {
-      throw new UnsupportedOperationException(s"Parameter $name not found")
-    })
-    if (wdlType != paramValue.wdlType) {
-      throw new UnsupportedOperationException(s"Incompatible type for $name: need a $wdlType, got a ${paramValue.wdlType}")
-    }
-    /* TODO: this also is a gross over-simplification */
-    paramValue.value.toString
-  }
+  def instantiate(parameters: Map[String, WdlValue]): Try[String] =
```

Comment:
The logic contained in this block is effectively:

    parameters.get(name).filter(_.wdlType == wdlType).map(_.value.toString)

which returns Option[String]

Related: Why is value Any? Is there any way to narrow that down?